src/App.js

```diff
@@ -16,46 +32,101 @@ const generateSquares = () => {
     for (let col = 0; col < 3; col += 1) {
       squares[row].push({
         id: currentId,
-        value: '',
+        value: "",
       });
       currentId += 1;
     }
   }
-
   return squares;
 }
 
 const App = () => {
-
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setPlayer] = useState(PLAYER_1)
+  const [winner, setWinner] = useState(null);
+  const [count, setCurrentCount] = useState(0)
 
   // Wave 2
   // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
-
+  //  When it is clicked on.
+  //  Then pass it into the squares as a callback
+  const onClickCallback = (id) => {
+    if (winner === "PLAYER_1" || winner === "PLAYER_2") return
+    const newSquare = [...squares] 
+    let i = 0
+    while(i < squares.length){
+      let x = 0
+      while (x < squares.length){
+        if (squares[i][x].id === id && squares[i][x].value === "" ){
+          squares[i][x].value = currentPlayer? PLAYER_1 : PLAYER_2;
+          setPlayer(!currentPlayer)
+          setWinner(checkForWinner())
+          setCurrentCount(count + 1)
+        };    
+        x += 1
+      }
+      i += 1
+    }
+   setSquares(newSquare);
+  }
 
+  // Complete in Wave 3
   const checkForWinner = () => {
-    // Complete in Wave 3
+    const newSquare = [];
+    squares.map(row => 
+      row.map(square => ( 
```

Comment:
Instead of using `.map` here, I'd use `forEach` since you're not converting the `squares` array into something else.